[xattrs] test_document_resurrection and test_sg_sdk_interop_unique_docs failing with NotFoundError

### Sync Gateway version
1.5.0-538

### Operating system
CentOS7

### Config file
```
{
    "out.stdout": {
        "adminInterface": "0.0.0.0:4985", 
        "cluster_config": {
            "bucket": "data-bucket", 
            "data_dir": ".", 
            "password": "password", 
            "server": "http://192.168.33.20:8091", 
            "username": "data-bucket"
        }, 
        "compressResponses": false, 
        "databases": {
            "db": {
                "bucket": "data-bucket", 
                "channel_index": {
                    "bucket": "index-bucket", 
                    "num_shards": 16, 
                    "password": "password", 
                    "server": "http://192.168.33.20:8091", 
                    "username": "index-bucket", 
                    "writer": true
                }, 
                "enable_shared_bucket_access": true, 
                "password": "password", 
                "server": "http://192.168.33.20:8091", 
                "username": "data-bucket"
            }
        }, 
        "interface": ":4984", 
        "log": [
            "CRUD+", 
            "Cache+", 
            "HTTP+", 
            "Changes+"
        ], 
        "maxCouchbaseConnections": 16, 
        "maxFileDescriptors": 90000, 
        "maxIncomingConnections": 0, 
        "slowServerCallWarningThreshold": 500
    }
}
```

### Log output
http://uberjenkins.sc.couchbase.com:8080/job/cen7-sync-gateway-functional-tests-base-di/838/testReport/testsuites.syncgateway.functional.tests/test_crud/test_document_resurrection_sync_gateway_default_functional_tests_tombstone_/

### Expected behavior
Test should pass

### Actual behavior
Test fails trying to verify sdk deletes. Test passes without xattrs.
```
>           verify_sdk_deletes(sdk_client, deleted_docs_to_verify)
testsuites/syncgateway/functional/tests/test_crud.py:180: 

>               sdk_client.get(doc_id)
E               Failed: DID NOT RAISE <class 'couchbase.exceptions.NotFoundError'>
```
### Steps to reproduce
`pytest -s -k test_document_resurrection --xattrs --ci --mode=di --server-version=5.0.0-3490 --sync-gateway-version=1.5.0-538 testsuites/syncgateway/functional/tests/`

--- a/base/bucket_gocb.go
+++ b/base/bucket_gocb.go
@@ -1436,14 +1436,15 @@ func (bucket CouchbaseBucketGoCB) WriteUpdateWithXattr(k string, xattrKey string
 		}
 
 		// Invoke callback to get updated value
-		updatedValue, updatedXattrValue, deleteDoc, err := callback(value, xattrValue, cas)
+		updatedValue, updatedXattrValue, isDelete, err := callback(value, xattrValue, cas)
 
 		if err != nil {
 			return emptyCas, err
 		}
 
-		// Attempt to write the updated document to the bucket
-		casOut, writeErr := bucket.WriteWithXattr(k, xattrKey, exp, cas, updatedValue, updatedXattrValue, deleteDoc)
+		// Attempt to write the updated document to the bucket.  Mark body for deletion if previous body was non-empty
+		deleteBody := len(value) > 0
+		casOut, writeErr := bucket.WriteWithXattr(k, xattrKey, exp, cas, updatedValue, updatedXattrValue, isDelete, deleteBody)
 		switch writeErr {
 		case nil:
 			return casOut, nil
@@ -1464,10 +1465,9 @@ func (bucket CouchbaseBucketGoCB) WriteUpdateWithXattr(k string, xattrKey string
 
 // Single attempt to update a document and xattr.  Setting isDelete=true and value=nil will delete the document body.  Both
 // update types (UpdateXattr, WriteCasWithXattr) include recoverable error retry.
-func (bucket CouchbaseBucketGoCB) WriteWithXattr(k string, xattrKey string, exp int, cas uint64, value []byte, xattrValue []byte, isDelete bool) (casOut uint64, err error) {
+func (bucket CouchbaseBucketGoCB) WriteWithXattr(k string, xattrKey string, exp int, cas uint64, value []byte, xattrValue []byte, isDelete bool, deleteBody bool) (casOut uint64, err error) {
 	// If this is a tombstone, we want to delete the document and update the xattr
 	if isDelete {
-		deleteBody := len(value) > 0
 		return bucket.UpdateXattr(k, xattrKey, exp, cas, xattrValue, deleteBody)
 	} else {
 		// Not a delete - update the body and xattr

--- a/rest/import_test.go
+++ b/rest/import_test.go
@@ -118,6 +118,48 @@ func TestXattrImportOldDoc(t *testing.T) {
 
 }
 
+// Validate tombstone w/ xattrs
+func TestXattrSGTombstone(t *testing.T) {
+
+	SkipImportTestsIfNotEnabled(t)
+
+	rt := RestTester{SyncFn: `
+		function(doc, oldDoc) { channel(doc.channels) }`}
+	defer rt.Close()
+
+	bucket := rt.Bucket()
+
+	rt.SendAdminRequest("PUT", "/_logging", `{"Import+":true, "CRUD+":true}`)
+
+	// 1. Create doc through SG
+	key := "TestXattrSGTombstone"
+	docBody := make(map[string]interface{})
+	docBody["test"] = key
+	docBody["channels"] = "ABC"
+
+	response := rt.SendAdminRequest("PUT", fmt.Sprintf("/db/%s", key), `{"channels":"ABC"}`)
+	assert.Equals(t, response.Code, 201)
+	log.Printf("insert response: %s", response.Body.Bytes())
+	var body db.Body
+	json.Unmarshal(response.Body.Bytes(), &body)
+	assert.Equals(t, body["ok"], true)
+	revId := body["rev"].(string)
+
+	// 2. Delete the doc through SG
+	response = rt.SendAdminRequest("DELETE", fmt.Sprintf("/db/%s?rev=%s", key, revId), "")
+	assert.Equals(t, response.Code, 200)
+	log.Printf("delete response: %s", response.Body.Bytes())
+	json.Unmarshal(response.Body.Bytes(), &body)
+	assert.Equals(t, body["ok"], true)
+	revId = body["rev"].(string)
+
+	// 3. Attempt to retrieve the doc through the SDK
+	deletedValue := make(map[string]interface{})
+	_, err := bucket.Get(key, deletedValue)
+	assertTrue(t, err != nil, "Expected key not found error trying to retrieve document")
+
+}
+
 // Test cas failure during WriteUpdate, triggering import of SDK write.
 // Disabled, as test depends on artificial latency in PutDoc to reliably hit the CAS failure on the SG write.  Scenario fully covered
 // by functional test.

--- a/test_integration.sh
+++ b/test_integration.sh
@@ -209,7 +209,8 @@ declare -a arr=(
     "TestChangesIncludeConflicts",
     "TestGetRemoved",
     "TestGetRemovedAndDeleted",
-    "TestGetRemovedAsUser")
+    "TestGetRemovedAsUser",
+    "TestXattrSGTombstone")
 
 
 # --------------------------------- SG Accel tests -----------------------------------------

